packages/backend-function/src/factory.ts

```diff
@@ -234,3 +284,37 @@ const nodeVersionMap: Record<NodeVersion, Runtime> = {
   18: Runtime.NODEJS_18_X,
   20: Runtime.NODEJS_20_X,
 };
+
+const secretPaths: string[] = [];
```

Comment:
translateEnvironmentProp should not communicate with the AmplifyFunction class via a shared variable like this. It creates an implicit lifecycle between calling this function and instantiating the class.

Instead we should refactor such that translating env vars and generating policies for them are encapsulated in a single component. eg

```ts
class FunctionEnvironmentTranslator {
  constructor(
    private readonly functionEnvironmentProps: HydratedFunctionProps['environment'],
    private readonly backendSecretResolver: BackendSecretResolver,
  ) {
  // do existing stuff in translateEnvironmentProps and store results in private class members
}
  
  getEnvironmentRecord = () => {
    return this.environmentRecord
  }

  getSecretPolicyStatement = (): PolicyStatement => {
    // use private secretPaths instance member to compute policy statement
  }
}
```